added hierarchical LDA classification scripts

MihailBogojeski opened the pull request:
This pull request was created due to the suggestion to add the hierarchical LDA classifier scripts to the main toolbox.

Commit:
added hierarchical LDA classification scripts

BenjaminBlankertz commented on classification/apply_HLDA.m:
```diff
@@ -0,0 +1,61 @@
+function out = apply_HLDA(epo, ival, C)
+% TRAIN_LDA - linear discriminant analysis 
```
% TRAIN_HLDA - Hierarchical Linear Discriminant Analysis

BenjaminBlankertz replied:
% TRAIN_HLDA - Hierarchical Linear Discriminant Analysis

BenjaminBlankertz commented on classification/apply_HLDA.m:
```diff
@@ -0,0 +1,61 @@
+nSegments = length(C.seg);
+dims = size(epo.x);
+% convert interval values to indices
+ival_idx = find(epo.t == ival(1)) : find(epo.t == ival(2));
```
This is a bit problematic. epo.t might not include the exact values given (think, e.g., about the case of a sampling rate of 256Hz). Therefore you should do something like:
     ibeg= max([1 find(epo.t<=ival(1))]);
     iend= min([find(epo.t>=ival(2)) length(epo.t)]);
     ival_idx= ibeg:iend;
However, you could also use the function proc_selectIval.

BenjaminBlankertz commented on classification/train_HLDA.m:
```diff
@@ -0,0 +1,63 @@
+if isempty(ival)
+    ival = [epo.t(1), epo.t(end)];
+end
+% convert interval values to indices
```
Use the function proc_selectIval.

BenjaminBlankertz commented on classification/apply_HLDA.m:
```diff
@@ -0,0 +1,61 @@
+%Description:
+%   APPLY_HLDA applies a hierarchical LDA classifier given an epo structure,
+%               a training interval and a number of segments
+%
```
Add reference to the paper describing HLDA.

BenjaminBlankertz commented on classification/apply_HLDA.m:
```diff
@@ -0,0 +1,61 @@
+function out = apply_HLDA(epo, ival, C)
```
If the apply function has arguments other than (fv, C) -- such as 'ival' here -- they cannot be used in crossvalidation. We should discuss, whether it makes sense to change this here.

MIkioBo replied:
We could get rid of the ival altogether (both in the train and apply functions), and simply pass the final X (with dimensions (time * channels) X samples) and the labels Y (similar to train_LDA in the toolbox). In this case, the needed interval should already be selected before passing X to the training function. In this way, the arguments and the  epo.t problems should be solved.

MihailBogojeski commented:
We could probably leave out the parameter ival altogether, and build the signature in a similar way to the regular train_LDA function, i.e., we would require X ( with dimensions (time \* channels) X samples) and Y (the labels) to be passed directly. So the selection of the relevant interval should already be done before X is passed. In this way, we avoid both the problem with the crossvalidation and the selection of the interval using epo.t.
EDIT: It is better for X to have three dimensions (time X channels X samples), because otherwise the segmentation may separate two groups of channels for the same time point in different segments.

Commit:
Changed signatures of both functions so ival is not used anymore.
Signature of apply_HLDA should now be compatible with cross validation.
Added optional parameter 'Regression' in train_HLDA, which can be used to specify
logistic regression as the top-level classifier for HLDA.
